### PR TITLE
feat(fal): add dbt 1.5 support 

### LINKED ISSUE #846: fal is not compatible with dbt 1.5.0
We upgraded the dbt version to 1.5.0, but when this upgrade finished it says that the version of dbt is not compatible with fal (0.8.6)

### PR DESCRIPTION
**Pull request by chamini2**
- snowflake
- bigquery
- postgres

*****

https://github.com/dbt-labs/dbt-core/issues/7465 is making it hard to reliably know where to search for artifacts (e.g. run_results) so we are still using the deprecated target-path in dbt_project.yml for pytest and we may need to use it for integration tests

closes #846 
closes FEA-1037

### CODE REVIEW

File: projects/fal/integration_tests/projects/000_fal_run/fal_scripts/after.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 output = output + f"\nModel alias is {model_alias}"
 alias_no_namespace = model_alias.split('__ns__')[1]
 output = output + f"\nModel alias without namespace is {alias_no_namespace}"
-temp_dir = os.environ["temp_dir"]
+temp_dir = os.getenv("temp_dir", ".")
```
Comment (mederka) [inline projects/fal/integration_tests/projects/000_fal_run/fal_scripts/after.py:21]: 👍

File: projects/fal/src/fal/cli/flow_runner.py [hunk 2]
```diff
@@ -47,10 +47,8 @@ def run_threaded(
 
 
 def fal_flow_run(parsed: argparse.Namespace) -> int:
-    arg_vars = getattr(parsed, "vars", "{}")
-
     # fal-format Python models
-    generated_models = generate_python_dbt_models(parsed.project_dir, arg_vars)
+    generated_models = generate_python_dbt_models(parsed.project_dir, parsed.vars)
```
Comment (mederka) [inline projects/fal/src/fal/cli/flow_runner.py:51]: When are we dropping these?

File: projects/fal/src/faldbt/parse.py [hunk 3]
```diff
@@ -89,6 +89,13 @@ def get_dbt_config(
         os.chdir(owd)
     else:
         config = RuntimeConfig.from_args(args)
+
+    # HACK: issue in dbt-core 1.5.0 https://github.com/dbt-labs/dbt-core/issues/7465
+    env_target_path = os.getenv("DBT_TARGET_PATH")
+    if env_target_path:
+        config.target_path = env_target_path
```
Comment (mederka) [inline projects/fal/src/faldbt/parse.py:96]: Will target changes require any action from current users that want to update to 1.5?

**Review by mederka — APPROVED** [review]
I think this looks great! Really awesome how things just seem to be tidier.

Great job, @chamini2 👏

File: projects/fal/src/fal/cli/flow_runner.py (hunk 2 above)
Comment (chamini2) [reply projects/fal/src/fal/cli/flow_runner.py:51]: > When are we dropping these?

Did not really underrstand the question.

I just introduced `--vars` support for `fal run` too (wanted to use vars for a different use and added it). But it's really not related to dbt 1.5

**Pull request merged**